### PR TITLE
Require name and MCI-fields for ac

### PR DESCRIPTION
**Pull request by gigxz**
Address [qa](https://www.pivotaltracker.com/story/show/185863108/comments/238386659) and [sentry](https://green-river.sentry.io/issues/4473350436/?alert_rule_id=12523843&alert_type=issue&notification_uuid=3da953eb-ae7b-4d3b-ac78-513ae6fb582b&project=4503977346662400&referrer=slack).

It would be more ideal if we could use the FormDefinition to specify the required-ness of these fields, but its complicated because (1) the logic for whether clearance is required or not is not readily available to the form, and (2) the name input component is grouped into 1 so you can't specify which name fields are required using the form. We really should address (2), but, this is a quick fix.

### CODE REVIEW

File: drivers/hmis_external_apis/extensions/hmis/hud/client_extension.rb [hunk 1]
```diff
@@ -49,23 +49,35 @@ def self.search_by_external_id(where, text)
             where.or(arel_table[:id].in(client_ids))
           end
 
+          private def requires_mci_clearance?
+            # If brand new Client record (NOT in context of an Enrollment), then MCI clearance is required
```
Comment (eanders) [inline drivers/hmis_external_apis/extensions/hmis/hud/client_extension.rb:66]: Does this need a "return unless MCI is available?"

**Review by eanders — APPROVED** [review]

**Comment by gigxz** [comment]
@eanders I looked at this with fresh eyes this morning, and unfortunately it had to be a lot more complicated so that we should show the right validations in the right contexts. I added comprehensive test cases to cover everything.

**Comment by gigxz** [comment]
I can't repro this test failure from ci😢  coming back to it

**Comment by eanders** [comment]
This latest batch of failures was a typo I introduced earlier today.  Hopefully that's fixed with the update from `release-86`

**Review by eanders — APPROVED** [review]
Thanks for the massive tests!

**Comment by eanders** [comment]
Looks like a couple of the tests aren't passing.

**Comment by gigxz** [comment]
yeah thats the one I couldn't repro yet..seems like it should be obvious but something must be different on my local

**Pull request merged**